app/uk/gov/hmrc/selfservicetimetopay/controllers/CalculatorController.scala

```diff
@@ -140,10 +140,13 @@ class CalculatorController(calculatorConnector: CalculatorConnector) extends Tim
 
   def submitCalculateInstalments: Action[AnyContent] = Action.async { implicit request =>
     sessionCache.get.flatMap[Result] {
-      case Some(ttpSubmission@TTPSubmission(Some(schedule), _, _, _, Some(_), Some(_), cd@CalculatorInput(debits, paymentToday, _, _, _, _), _)) =>
+      case Some(ttpSubmission@TTPSubmission(Some(schedule), _, _, taxpayer@Some(_), Some(_), _, cd@CalculatorInput(debits, paymentToday, _, _, _, _), _)) =>
```

Comment:
We cannot assume that we have a Taxpayer plus this is being mapped to a Some(_) which would not catch a None case for Taxpayer. As a result, a situation without taxpayer is not being handled correctly.